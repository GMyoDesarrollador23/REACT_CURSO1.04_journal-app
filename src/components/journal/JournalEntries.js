import React from "react";
import { useSelector } from "react-redux";
import { JornalEntry } from "./JornalEntry";

export const JournalEntries = () => {
   const { notes } = useSelector((state) => state.notes);

   return (
      <div className="journal__entries ">
         {notes.map((note) => (
            <JornalEntry key={note.id} {...note} />
         ))}
      </div>
   );
};
