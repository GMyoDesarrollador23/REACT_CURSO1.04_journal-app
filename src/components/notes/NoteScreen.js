import Swal from "sweetalert2";
import React, { useEffect, useRef } from "react";
import { useSelector, useDispatch } from "react-redux";

import { NoteAppBar } from "./NoteAppBar";
import { useForm } from "../../hooks/useForm";
import {
   activeNote,
   startDeleting,
} from "../../actions/notes";

export const NoteScreen = () => {
   const dispatch = useDispatch();

   const { active: note } = useSelector(
      (state) => state.notes
   );

   const [formValues, handleInputChange, reset] = useForm(
      note
   );
   const { title, body, id } = formValues;

   const activeId = useRef(note.id);

   useEffect(() => {
      if (note.id !== activeId.current) {
         reset(note);
         activeId.current = note.id;
      }
   }, [note, reset]);

   useEffect(() => {
      dispatch(
         activeNote(formValues.id, { ...formValues })
      );
   }, [formValues, dispatch]);

   const handleDelete = () => {
      Swal.fire({
         title: "Delete Note",
         text: "Are you sure you want to delete this note?",
         icon: "warning",
         showCancelButton: true,
         confirmButtonColor: "#3085d6",
         cancelButtonColor: "#d33",
         confirmButtonText: "Yes, delete it!",
      }).then((result) => {
         if (result.value) {
            dispatch(startDeleting(id));
            Swal.fire({
               position: "center",
               icon: "success",
               title: "note cleared with success",
               showConfirmButton: false,
               timer: 1500,
            });
         }
      });
   };

   return (
      <div className="notes__main-content animate__animated animate__fadeIn animate__faster">
         <NoteAppBar />

         <div className="notes__content">
            <input
               type="text"
               placeholder="hola"
               className="notes__title-input"
               autoComplete="off"
               name="title"
               value={title}
               onChange={handleInputChange}
            />
            <textarea
               placeholder="que paso ayer"
               className="notes__texaria"
               name="body"
               value={body}
               onChange={handleInputChange}
            ></textarea>
            {note.url && (
               <div className="notes__image animate__animated animate__fadeIn animate__faster">
                  <img src={note.url} alt="paisaje" />
               </div>
            )}
         </div>
         <button
            className="btn btn-danger"
            onClick={handleDelete}
         >
            Delete
         </button>
      </div>
   );
};
