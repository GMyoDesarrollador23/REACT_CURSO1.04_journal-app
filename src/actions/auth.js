import Swal from "sweetalert2";

import { types } from "../types/types";
import {
   firebase,
   googleAuthProvider,
} from "../firebase/firebase-config";
import { finishLoading, startLoading } from "./ui";

import { notesLogout } from "../actions/notes";

export const startLoginEmailPassword = (
   email,
   password
) => {
   return (dispatch) => {
      dispatch(startLoading());
      return firebase
         .auth()
         .signInWithEmailAndPassword(email, password)
         .then(({ user }) => {
            dispatch(login(user.uid, user.displayName));

            dispatch(finishLoading());
            Swal.fire({
               position: "center",
               icon: "success",
               title: `welcome back "${user.displayName}"`,
               showConfirmButton: false,
               timer: 2000,
            });
         })
         .catch((e) => {
            console.warn(e);
            dispatch(finishLoading());
            Swal.fire({
               position: "center",
               icon: "error",
               title: e.message,
               showConfirmButton: false,
               timer: 2000,
            });
         });
   };
};

export const startRegisterWithEmailPasswordName = (
   email,
   password,
   name
) => {
   return (dispatch) => {
      firebase
         .auth()
         .createUserWithEmailAndPassword(email, password)
         .then(async ({ user }) => {
            await user.updateProfile({
               displayName: name,
            });
            // console.log(user);

            dispatch(login(user.uid, user.displayName));
         })
         .catch((e) => {
            console.log(e);
            Swal.fire({
               position: "center",
               icon: "error",
               title: e.message,
               showConfirmButton: false,
               timer: 2000,
            });
         });
   };
};

export const startGoogleLogin = () => {
   return (dispatch) => {
      firebase
         .auth()
         .signInWithPopup(googleAuthProvider)
         .then(({ user }) => {
            dispatch(login(user.uid, user.displayName));
         });
   };
};

export const startLogoud = () => {
   return async (dispatch) => {
      await firebase.auth().signOut();

      dispatch(logout());
      dispatch(notesLogout());
   };
};

export const login = (uid, displayName) => ({
   type: types.login,
   payload: {
      uid,
      displayName,
   },
});

export const logout = () => ({
   type: types.logout,
});
