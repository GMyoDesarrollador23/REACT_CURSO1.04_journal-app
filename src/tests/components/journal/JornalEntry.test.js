import React from "react";
import { mount } from "enzyme";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";

import thunk from "redux-thunk";
import configureStore from "redux-mock-store";

import { activeNote } from "../../../actions/notes";
import { JornalEntry } from "../../../components/journal/JornalEntry";

// *****************************************************

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
   id: 123,
   date: 0,
   title: "hola",
   body: "Mundo",
   url: "https://hola/mundo.jpg",
};

const wraper = mount(
   <Provider store={store}>
      <JornalEntry {...note} />
   </Provider>
);

// *****************************************************

describe("pruebas en <JornalEntry/>", () => {
   // -------------------------------------------
   test("debe de mostrarse correctamente", () => {
      expect(wraper).toMatchSnapshot();
   });
   // -------------------------------------------
   test("debe de activar la nota", () => {
      wraper.find(".journal__entry").prop("onClick")();
      expect(store.dispatch).toHaveBeenCalledWith(
         activeNote(note.id, { ...note })
      );
   });
   // -------------------------------------------
});
