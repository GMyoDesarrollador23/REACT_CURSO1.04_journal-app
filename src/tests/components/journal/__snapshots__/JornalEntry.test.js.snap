// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <JornalEntry/> debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry  pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://hola/mundo.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      hola
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal__entry-dat e-box"
  >
    <span>
      12/21/1969
    </span>
  </div>
</div>
`;
